RHOARDOC-1606: Document Vert.x Spring Boot reactive examples

inoxx03 opened the pull request.

tradej commented on docs/topics/assembly_developing-and-deploying-spring-reactive-application.adoc:
```diff
@@ -0,0 +1,58 @@
+[id='developing-reactive-spring-boot-applications-webflux']
```
{context} (in all files)

inoxx03 replied:
Thanks, added `{context}` to all the procedure IDs and the assembly ID.

gytis commented on docs/topics/assembly_developing-and-deploying-spring-reactive-application.adoc:
```diff
@@ -0,0 +1,58 @@
+[id='developing-reactive-spring-boot-applications-webflux']
+= Developing reactive {SpringBoot} applications using Spring Reactive and Webflux
```
WebFlux is relevant only for a subset of reactive applications. Unless we plan to have a different section which will talk about mail, AMQP, Kafka and whatever else will come in the future, we shouldn't have WebFlux in a main title.
Assuming the project name stays, the title could be "Developing reactive applications with Vert.x Spring Boot"?
What do you think @nainaz

inoxx03 replied:
Ah, OK, initial understanding was that all of these examples would be based on WebFlux.
Since this not the case, I agree that the title should be _Developing reactive applications with Vert.x Spring Boot_.

gytis replied:
At the moment all these samples use WebFlux. However, the project itself is not limited to it, e.g. use can use email client on it's own. So to minimise confusion in the future once we expand the documentation, I would prefer to keep it more generic.

inoxx03 replied:
I've re-titled the section to _Developing reactive applications with Vert.x Spring Boot_.
Since I'me not sure whether _Vert.x Spring Boot_ will be the official name for the Spring+reactive GA release, I'll replace the name with an attribute so we can change it later.

inoxx03 replied:
UPDATE: According to my conversation with Naina, we should be using _SpringBoot with Vert.X_  as the name.

gytis commented on docs/topics/proc_creating-a-reactive-http-service-with-webflux.adoc:
```diff
@@ -0,0 +1,83 @@
+
+.Procedure
+
+. Add `vertx-spring-boot-starter` as a dependency in the `pom.xml` file of your project.
```
`vertx-spring-boot-starter-http`

inoxx03 replied:
thanks you:) fixed  in all instances

gytis commented on docs/topics/resources/spring-boot/reactive-examples/example-http-service-webflux-pom.xml:
```diff
@@ -0,0 +1,38 @@
+    <dependency>
+      <groupId>dev.snowdrop</groupId>
+      <artifactId>vertx-spring-boot-starter-http</artifactId>
+      <version>${project.version}</version>
```
I know this was taken from the sample. But because this documentation assumes that `spring-boot-bom` is used, `version` tag is not necessary here.
If this whole pom.xml is exposed somewhere, we should rewrite it to look the way it should from the documentation perspective.

inoxx03 replied:
The full POM files for examples are maintained in the docs repo, but are currently not exposed.
I'll rewrite the examples to remove the version property references, because the documented procedure also assumes that `spring-boot-bom` is imported in the POM file.

inoxx03 replied:
removed version references form POM examples

gytis commented on docs/topics/proc_creating-reactive-authenticated-http-service-webflux.adoc:
```diff
@@ -0,0 +1,128 @@
+
+.Procedure
+
+. Add `vertx-spring-boot-starter` and `spring-boot-starter-security` as dependencies in the `pom.xml` file of your project.
```
`vertx-spring-boot-starter-http`

gytis commented on docs/topics/proc_creating-reactive-authenticated-http-service-webflux.adoc:
```diff
@@ -0,0 +1,128 @@
+[id='creating-an-authenticated-reactive-spring-boot-http-service-with-webflux']
+= Creating an authenticated reactive {SpringBoot} HTTP service with WebFlux
```
This sample demonstrates basic authentication. Maybe rename it to "Using basic authentication with a reactive Spring Boot WebFlux application"

inoxx03 replied:
Yes, I've been thinking the same thing:) I'll go ahead and rename it.

inoxx03 replied:
Title changed

gytis commented on docs/topics/resources/spring-boot/reactive-examples/example-http-service-authenticated-webflux-pom.xml:
```diff
@@ -0,0 +1,42 @@
+    <dependency>
+      <groupId>dev.snowdrop</groupId>
+      <artifactId>vertx-spring-boot-starter-http</artifactId>
+      <version>${project.version}</version>
```
Same as in the simple HTTP sample. No need for the version because `spring-boot-bom` is expected by the documentation.

gytis commented on docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc:
```diff
@@ -0,0 +1,108 @@
+
+. Add the following dependencies in the `pom.xml` file of your project:
+
+* `vertx-spring-boot-starter`
```
`vertx-spring-boot-starter-http`

gytis commented on docs/topics/resources/spring-boot/reactive-examples/example-0auth2-with-reactive-http-application-webflux-pom.xml:
```diff
@@ -0,0 +1,51 @@
+    <dependency>
+      <groupId>dev.snowdrop</groupId>
+      <artifactId>vertx-spring-boot-starter-http</artifactId>
+      <version>${project.version}</version>
```
Same as in others, no need for version

inoxx03 replied:
Thank  you:) Fixed in all instances

gytis commented on docs/topics/proc_creating-reactive-smtp-email-app-webflux.adoc:
```diff
@@ -0,0 +1,92 @@
+
+.Procedure
+
+. Add `vertx-spring-boot-starter` and `vertx-spring-boot-starter-mail` as dependencies in the `pom.xml` file of your project.
```
`vertx-spring-boot-starter-http`

gytis commented on docs/topics/resources/spring-boot/reactive-examples/example-http-service-mail.pom.xml:
```diff
@@ -0,0 +1,43 @@
+    <dependency>
+      <groupId>dev.snowdrop</groupId>
+      <artifactId>vertx-spring-boot-starter-http</artifactId>
+      <version>${project.version}</version>
```
No need for version

gytis commented on docs/topics/resources/spring-boot/reactive-examples/example-http-service-mail.pom.xml:
```diff
@@ -0,0 +1,43 @@
+<!-- end::http-service-mail-pom-dependencies[] -->
+  </dependencies>
+
+  <build>
```
No need for version

gytis commented on docs/topics/proc_creating-reactive-smtp-email-app-webflux.adoc:
```diff
@@ -0,0 +1,92 @@
+
+. Create a mail handler class for your application: 
++
+.src/main/java/dev/snowdrop/vertx/sample/mail/MailHandler.java`
```
Source missing

inoxx03 replied:
Ah, My bad, sorry, I'll make sure to include it, but if you have  plans to re-work the mail example, I can wait:)

inoxx03 replied:
Added source form the refactored email app sample.

inoxx03 commented:
@gytis Thank you for the feedback and suggestions, I'll make sure to address all the issues you've pointed out.

to answer your question:
> At the moment the samples are only briefly presented but specific nuances are not explained e.g. different resource handling, security considerations, ssl mail etc. Do you plan to expand more in the future? Do you need my input on that?

Yes, I would definitely like to add more details about resource handling, application design tradeoffs, security considerations and possible caveats that apply to using the presented designs in production environments.
I will greatly appreciate input from you on these aspects of the example apps. Thank you.

inoxx03 commented:
@gytis thank you for the review. I've addressed most of your comments, Updated the email example and added a procedure for the Server-sent events example.
If this is acceptable as a docs MVP, then I'd like to start expanding the modules with conceptual information from the spring reactive docs sources that we discussed at our call.
Please let me know your thoughts. Thank you.

inoxx03 commented:
@huntley Please take a look and let me know what you think:) Thank you:)
